ramalama/model.py

```diff
@@ -13,7 +13,7 @@ class Model:
 
     def __init__(self, model):
         self.model = model
-        if sys.platform == "darwin":
+        if sys.platform == "darwin" or os.getenv("HIP_VISIBLE_DEVICES") or os.getenv("CUDA_VISIBLE_DEVICES"):
```

Comment:
When using a ROCm or CUDA GPU, in the case where there are multiple GPUs you can select one specifically using HIP_VISIBLE_DEVICES or CUDA_VISIBLE_DEVICES environment variable.

How do we select a GPU in vulkan/kompute @slp , just curious how we will adapt this for the kompute case...

Reply:
Should we allow users to select the GPU on the command line as well?

Reply:
Yeah there's lots of options like that we should/could add... Select a specific GPU type (AMD/Nvidia), select a specific GPU framework (ROCm/Nvidia/Vulkan/Kompute/OpenCL/etc.), select a specific GPU (AMD GPU 1, AMD GPU 2)... I just wonder should we do this up front or wait until someone requests the feature...

What we could also do is... If HIP_VISIBLE_DEVICES or CUDA_VISIBLE_DEVICES is already set externally don't look for the GPU with the most VRAM on the system...|

That's kinda the simple heuristic for now for auto-detecting the primary GPU, if we have a GPU that has more than 1GB VRAM, run on the GPU with the most VRAM...

But this PR only does it for AMD GPUs for now...

Reply:
I was thinking they would actually give us the device(s)
--gpu /dev/dri/card0 or something like that.

Reply:
The problem is, it's not always straightforward to infer what /dev/dri/card0 is Nvidia/AMD or something else

Reply:
/dev/dri/card0 is not deterministic also, so that should not be used in a quadlet for example. I fixed that bug in multiple graphics frameworks as a recent kernel change made card0, card1, etc. less deterministic than ever...

https://github.com/libsdl-org/SDL/pull/5865
https://swiftshader.googlesource.com/SwiftShader/+/afbfd351374d93484f8eecf532afbd0dac084865